drivers/timer/stm32_lptim_timer.c

```diff
@@ -0,0 +1,279 @@
+
+/* nb of LPTIM counter unit per kernel tick  */
+#define COUNT_PER_TICK (LPTIM_CLOCK / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+struct device;	       /* forward declaration; type is not used. */
+
+static volatile u32_t last_cnt;
+
+static struct k_spinlock lock;
+
+static void lptim_irq_handler(struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+	if ((LL_LPTIM_IsActiveFlag_ARRM(LPTIM1) != 0)
+		&& (LL_LPTIM_IsEnabledIT_ARRM)(LPTIM1) != 0) {
+
+		k_spinlock_key_t key = k_spin_lock(&lock);
+
+		/* LPTIM1 CNT could have already been reset */
+		u32_t cnt_value = LL_LPTIM_GetAutoReload(LPTIM1) + 1
+				+ LL_LPTIM_GetCounter(LPTIM1);
+
+		/* do not set ARR yet, z_clock_announce will do */
+
+		/* a next timeout will be raised */
+		LL_LPTIM_ClearFLAG_ARRM(LPTIM1);
+		/* reset last_cnt for a next call to 'elapsed' */
+		last_cnt = 0;
+
+		k_spin_unlock(&lock, key);
+
+		/* announce the elapsed time in ms, cnt_value is 16bit */
+		u32_t dticks = (cnt_value
+				* (CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+				/ (LPTIM_CLOCK);
+		z_clock_announce(IS_ENABLED(CONFIG_TICKLESS_KERNEL)
+			? dticks : 1);
+	}
+}
+
+int z_clock_driver_init(struct device *device)
+{
+	ARG_UNUSED(device);
+
+	/* enable LPTIM clock source */
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_LPTIM1);
+	LL_APB1_GRP1_ReleaseReset(LL_APB1_GRP1_PERIPH_LPTIM1);
+	/* select LSI as LPTIM1 clock source */
+
+#if defined(CONFIG_STM32_LPTIM_CLOCK_LSI)
+	/* enable LSI clock */
+	LL_RCC_LSI_Enable();
+	while (!LL_RCC_LSI_IsReady()) {
+		/* Wait for LSI ready */
+	}
+	LL_RCC_SetLPTIMClockSource(LL_RCC_LPTIM1_CLKSOURCE_LSI);
+
+#else /* CONFIG_STM32_LPTIM_CLOCK_LSI */
+#if defined(LL_APB1_GRP1_PERIPH_PWR)
+	/* Enable the power interface clock */
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
+#endif /* LL_APB1_GRP1_PERIPH_PWR */
+	/* enable backup domain */
+	LL_PWR_EnableBkUpAccess();
+	LL_RCC_ForceBackupDomainReset();
+	LL_RCC_ReleaseBackupDomainReset();
+
+	/* enable LSE clock */
+	LL_RCC_LSE_DisableBypass();
+	LL_RCC_LSE_Enable();
+	while (!LL_RCC_LSE_IsReady()) {
+		/* Wait for LSE ready */
+	}
+	LL_RCC_SetLPTIMClockSource(LL_RCC_LPTIM1_CLKSOURCE_LSE);
+
+#endif /* CONFIG_STM32_LPTIM_CLOCK_LSI */
+
+	/* Clear the event flag and possible pending interrupt */
+	IRQ_CONNECT(DT_LPTIM_1_IRQ, DT_LPTIM_1_IRQ_PRI,
+			lptim_irq_handler, 0, 0);
+	irq_enable(DT_LPTIM_1_IRQ);
+
+	/* configure the LPTIM1 counter */
+	LL_LPTIM_SetClockSource(LPTIM1, LL_LPTIM_CLK_SOURCE_INTERNAL);
+	/* configure the LPTIM1 prescaler with 1 */
+	LL_LPTIM_SetPrescaler(LPTIM1, LL_LPTIM_PRESCALER_DIV1);
+	LL_LPTIM_SetPolarity(LPTIM1, LL_LPTIM_OUTPUT_POLARITY_REGULAR);
+	LL_LPTIM_SetUpdateMode(LPTIM1, LL_LPTIM_UPDATE_MODE_IMMEDIATE);
+	LL_LPTIM_SetCounterMode(LPTIM1, LL_LPTIM_COUNTER_MODE_INTERNAL);
+	LL_LPTIM_DisableTimeout(LPTIM1);
+	/*LL_LPTIM_EnableResetAfterRead(LPTIM1);*/
+	LL_LPTIM_TrigSw(LPTIM1);
+#ifdef LL_LPTIM_INPUT1_SRC_GPIO
+	LL_LPTIM_SetInput1Src(LPTIM1, LL_LPTIM_INPUT1_SRC_GPIO);
```

Comment:
Can you elaborate why is this line required actually?

Reply:
this is the default input pin configuration

Reply:
So maybe this is not needed. We don' t use inputs anyway currently, so maybe this is clearer if we remove these.
.